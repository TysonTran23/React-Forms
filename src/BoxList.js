import React, { useState } from "react";

import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

import { v4 as uuid } from "uuid";

const BoxList = () => {
  const INITIAL_STATE = [
    { id: uuid(), color: "red", width: "100px", height: "100px" },
  ];

  const [boxes, setBoxes] = useState(INITIAL_STATE);

  const addBox = (newBox) => {
    setBoxes((boxes) => [...boxes, { ...newBox, id: uuid() }]);
  };

  const deleteBox = (id) => {
    setBoxes((boxes) => boxes.filter((box) => box.id !== id));
  };

  return (
    <div>
      <h1>Boxes</h1>
      <NewBoxForm addBox={addBox} />
      <div>
        {boxes.map(({ id, color, width, height }) => (
          <Box
            key={id}
            id={id}
            color={color}
            width={width}
            height={height}
            deleteBox={deleteBox}
          />
        ))}
      </div>
    </div>
  );
};

export default BoxList;
