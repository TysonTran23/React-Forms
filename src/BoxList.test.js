import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function () {
  render(<BoxList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

// it("should add a new box", function () {
//   const { queryByText, getByLabelText, queryAllByText } = render(<BoxList />);
//   const colorInput = getByLabelText("Color:");
//   const widthInput = getByLabelText("Width:");
//   const heightInput = getByLabelText("Height:");
//   const button = queryByText("Add Box!");
//   fireEvent.change(colorInput, { target: { value: "blue" } });
//   fireEvent.change(widthInput, { target: { value: "100px" } });
//   fireEvent.change(heightInput, { target: { value: "100px" } });
//   fireEvent.click(button);

//   expect(queryAllByText("X")).toBeInTheDocument();
// });
