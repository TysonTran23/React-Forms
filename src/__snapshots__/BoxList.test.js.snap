// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Boxes
    </h1>
    <form>
      <label
        for="color"
      >
        Color: 
      </label>
      <input
        id="color"
        name="color"
        placeholder="Ex: red"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        id="width"
        name="width"
        placeholder="100px"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height: 
      </label>
      <input
        id="height"
        name="height"
        placeholder="100px"
        type="text"
        value=""
      />
      <button>
        Add Box!
      </button>
    </form>
    <div>
      <div
        style="background-color: red; width: 100px; height: 100px;"
      />
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
